// zingchart-angular/esm2015/lib/zingchart-angular.service.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/zingchart-angular.service.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
export class ZingchartAngularService {
    constructor() {
        this.count = 0;
    }
    /**
     * @return {?}
     */
    getCount() {
        return this.count;
    }
    /**
     * @return {?}
     */
    increment() {
        this.count++;
    }
    /**
     * @return {?}
     */
    decrement() {
        this.count--;
    }
}
ZingchartAngularService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] }
];
/** @nocollapse */
ZingchartAngularService.ctorParameters = () => [];
/** @nocollapse */ ZingchartAngularService.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function ZingchartAngularService_Factory() { return new ZingchartAngularService(); }, token: ZingchartAngularService, providedIn: "root" });
if (false) {
    /**
     * @type {?}
     * @private
     */
    ZingchartAngularService.prototype.count;
}
