// zingchart-angular/esm5/lib/zingchart-angular.service.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/zingchart-angular.service.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
var ZingchartAngularService = /** @class */ (function () {
    function ZingchartAngularService() {
        this.count = 0;
    }
    /**
     * @return {?}
     */
    ZingchartAngularService.prototype.getCount = /**
     * @return {?}
     */
    function () {
        return this.count;
    };
    /**
     * @return {?}
     */
    ZingchartAngularService.prototype.increment = /**
     * @return {?}
     */
    function () {
        this.count++;
    };
    /**
     * @return {?}
     */
    ZingchartAngularService.prototype.decrement = /**
     * @return {?}
     */
    function () {
        this.count--;
    };
    ZingchartAngularService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    ZingchartAngularService.ctorParameters = function () { return []; };
    /** @nocollapse */ ZingchartAngularService.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function ZingchartAngularService_Factory() { return new ZingchartAngularService(); }, token: ZingchartAngularService, providedIn: "root" });
    return ZingchartAngularService;
}());
export { ZingchartAngularService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    ZingchartAngularService.prototype.count;
}
